### PR TITLE
Created composer.json

### PR DESCRIPTION
**Pull request by spacedmonkey**
Added the composer.json file, to make this github repo compatible with composer.

### CODE REVIEW

**Comment by tommymarshall** [comment]
Would love to have this rather than relying on a fork.

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"type"       : "wordpress-plugin",
+	"license"    : "GPL-2.0+",
+	"authors"    : [
+		{
+			"name"    : "Elliot Condon",
+			"homepage": "http://www.elliotcondon.com/"
+		}
+	],
+	"require": {
+          "composer/installers": "~1.0"
```
Comment (thaiphan) [inline composer.json]: Are you missing a closing bracket here?

**Comment by spacedmonkey** [comment]
Any chance this is going to be merged?

**Comment by elliotcondon** [comment]
HI guys

Thanks for the addition, can you please explain what this does, and how it benefits the plugin?
Forgive me, but I don't know anything about composer

**Comment by kalenjohnson** [comment]
This seems related: http://support.advancedcustomfields.com/forums/topic/composer-support-for-premium-add-ons/

@elliotcondon you had said:
> Just wanted to jump in and let you all know that ACF v5 pro will be compatible with composer!
> I’ll even write up a tutorial for it in the near future.

**Comment by austinpray** [comment]
@elliotcondon we have a good blog post here: https://roots.io/using-composer-with-wordpress/

**Comment by elliotcondon** [comment]
Hi guys

Thanks for the replies. I must admit, I don't remember posting what was quoted, but that aside, I'm still not understanding what composer will do, and how it will work for a premium plugin which does not have a public download URL.

**Comment by elliotcondon** [comment]
I guess i'm just wanting to understand what will happen when I merge in this PR? What new functionality will be added, and how will bit be used?

Thanks mate

**Comment by austinpray** [comment]
>I guess i'm just wanting to understand what will happen when I merge in this PR?

Another blog post! https://roots.io/wordpress-plugins-with-composer/

**Comment by KLVTZ** [comment]
@elliotcondon perhaps I can add a direct answer to this. By adding a composer file you can take advantage of the popular PHP package repository: https://packagist.org/. By making your plugin available in this manner, we can take advantage of a few benefits.

If you notice in the above PR, it includes defining the plugin as a type of `wordpress-plugin`. Doing so would allow consumers to grab the repo and install it to a custom directory. This is thanks to the package requirement above that would use https://github.com/composer/installers. In projects, like Bedrock, they allow developers to define custom folders for plugins to live in.

To see the implementation above in action, check out FoundationPress
https://github.com/olefredrik/FoundationPress/pull/392

**Comment by elliotcondon** [comment]
Thanks guys, I've merged in this PR.
Sorry for the time taken, tonnes of work going on at the moment!!!

Please post back with any changes needed, and also how this has helped.
- real world examples are much easier for me to understand than reading theoretical blog posts :)

**Comment by austinpray** [comment]
@elliotcondon real world application of composer + WordPress https://github.com/roots/bedrock

**Comment by KLVTZ** [comment]
@elliotcondon yoast is always a good example: https://github.com/Yoast/wordpress-seo

Autoloading can also be made available in a project. Therefore, we don't have to require an object in a wordpress theme to use. Instead, we can have the object autoloaded and readily available to use:
https://github.com/KLVTZ/offgrid-image

**Comment by pea** [comment]
Any news on when ACF might be added to Packagist?

**Comment by KLVTZ** [comment]
@pea check out https://wpackagist.org/ and search for `advanced-custom-fields`

**Comment by pea** [comment]
@KLVTZ Thanks. Is the pro version a part of the advanced-custom-fields package on that?

**Comment by KLVTZ** [comment]
@pea No, the above package is only the standard edition. If you want to incorporate the paid version into composer, you would have to create your own private repository (so the free version isn't leaked) and maintain it under `composer.json`. You can find more info here:

https://getcomposer.org/doc/05-repositories.md#vcs

Another thing I would suggest is using https://github.com/composer/installers as a way to install a package to the correct folder (not the vendor folder but the plugins folder).

Probably the best summary of this flow would be:

https://roots.io/using-composer-with-wordpress/

Hope this helps!

**Pull request merged**